uniqueid: compilation error on **other** boards

#### Description

Trying to compile an error is raised building `sys/uniqueid` component

#### Steps to reproduce the issue
execute:
```sh
make BOARD=cc1352p-launchpad
```
#### Expected results
Compilation success

#### Actual results
```sh
"make" -C ./m4a-firmware/firmware/sys/uniqueid
/usr/lib/gcc/arm-none-eabi/9.2.1/../../../arm-none-eabi/bin/ld: 
.../m4a-firmware/firmware/bin/cc1312-launchpad/periph_common/init.o: in function `periph_init':
..//m4a-firmware/RIOT/drivers/periph_common/init.c:101: undefined reference to `hwrng_init'
```
#### Versions
OS: Ubuntu 22.04
RIOT: project current version

--- a/firmware/sys/uniqueid/Kconfig
+++ b/firmware/sys/uniqueid/Kconfig
@@ -20,8 +20,10 @@ menu "Subnet"
         config MODE_STATIC
             bool "STATIC"
 
+        if USEMODULE_AT86RF2XX || USEMODULE_AT86RF215 || USEMODULE_PERIPH_HWRNG
         config MODE_RANDOM
             bool "RANDOM"
+        endif
 
         config MODE_MANUAL
             bool "MANUAL"

--- a/firmware/sys/uniqueid/Makefile.dep
+++ b/firmware/sys/uniqueid/Makefile.dep
@@ -4,8 +4,10 @@ USEMODULE += netdev_default
 USEMODULE += gnrc_ipv6_default
 USEMODULE += auto_init_gnrc_netif
 
-ifneq (,$(filter m4a-24g m4a-mb,$(BOARD)))
+ifneq (,$(filter at86rf2xx at86rf215,$(USEMODULE)))
     USEMODULE += radio
 else
-    USEMODULE += periph_hwrng
+    ifneq (,$(filter periph_hwrng,$(HAS_PERIPH_HWRNG)))
+        USEMODULE += periph_hwrng
+    endif
 endif

--- a/firmware/sys/uniqueid/uniqueid.c
+++ b/firmware/sys/uniqueid/uniqueid.c
@@ -29,7 +29,7 @@
 #include "net/gnrc.h"
 #if MODULE_AT86RF2XX || MODULE_AT86RF215
 #include "radio.h"
-#else
+#elif MODULE_PERIPH_HWRNG
 #include "periph/hwrng.h"
 #endif
 void subnet_to_ipv6(ipv6_addr_t *addr) {

--- a/tests/uniqueid/main.c
+++ b/tests/uniqueid/main.c
@@ -42,18 +42,23 @@ void get_unique_from_mac(ipv6_addr_t *output) {
 }
 
 void test_get_ipv6Address(void) {
+#if CONFIG_MODE_STATIC || CONFIG_MODE_MANUAL
     ipv6_addr_t ipv6 = {
         .u8 = {0},
     };
-
-    subnet_to_ipv6(&ipv6);
-#ifdef CONFIG_MODE_STATIC
     ipv6_addr_t output = {
         .u8 = {0},
     };
+    printf("\n");
+    subnet_to_ipv6(&ipv6);
     ipv6_addr_print(&ipv6);
+    printf("\n");
     get_unique_from_mac(&output);
+#if CONFIG_MODE_STATIC
     TEST_ASSERT_EQUAL_INT(1, ipv6_addr_equal(&ipv6, &output));
+#elif CONFIG_MODE_MANUAL
+    TEST_ASSERT_EQUAL_INT(1, !ipv6_addr_equal(&ipv6, &output));
+#endif
 #endif
 
 #ifdef CONFIG_MODE_RANDOM
@@ -63,7 +68,6 @@ void test_get_ipv6Address(void) {
     ipv6_addr_t output2 = {
         .u8 = {0},
     };
-
     subnet_to_ipv6(&output1);
     subnet_to_ipv6(&output2);
     printf("\nFirst random IPv6\n");

